superset-frontend/src/components/FilterableTable/FilterableTable.tsx

```diff
@@ -323,19 +323,19 @@ export default class FilterableTable extends PureComponent<
 
   sortResults(sortBy: string, descending: boolean) {
     return (a: Datum, b: Datum) => {
-      const aValue = a[sortBy];
-      const bValue = b[sortBy];
-      if (aValue === bValue) {
-        // equal items sort equally
-        return 0;
-      }
-      if (aValue === null) {
-        // nulls sort after anything else
-        return 1;
-      }
-      if (bValue === null) {
-        return -1;
-      }
+      // Parse any floating numbers so they'll sort correctly
+      const parseFloatingNums = (value: any) => {
+        const floatValue = parseFloat(value);
+        return Number.isNaN(floatValue) ? value : parseFloat(value);
+      };
```

Comment:
I think it's better to move this function outside of `sortResults`, otherwise it gets defined for every comparison when sorting the array.

Reply:
I overlooked this comment, looking into this now

Reply:
Good call, moved the function in [`this commit`](https://github.com/apache/superset/pull/17573/commits/628a5493439e227452c2805aff5025a885593bcb).